xonsh/history/json.py

```diff
@@ -15,53 +15,70 @@
 import xonsh.xoreutils.uptime as uptime
 
 
-def _xhj_gc_commands_to_rmfiles(hsize, files):
-    """Return the history files to remove to get under the command limit."""
-    rmfiles = []
+def _xhj_gc_commands_to_rmfiles(hsize, files) -> ([], int):
+    """Return number of units and list of history files to remove to get under the limit,
+
+    Parameters:
+    -----------
+    hsize (int):  units of history, # of commands in this case.
+    files ((mod_ts, num_commands, path)[], fsize): history files, sorted oldest first.
+
+    Returns:
+    --------
+    hsize_removed (int):  units of history to be removed
+    rm_files ((mod_ts, num_commands, path, fsize)[]): list of files to remove.
+    """
     n = 0
     ncmds = 0
-    for ts, fcmds, f in files[::-1]:
-        if fcmds == 0:
-            # we need to make sure that 'empty' history files don't hang around
-            rmfiles.append((ts, fcmds, f))
+    for _, fcmds, f, _ in files[::-1]:
```

Comment:
should be `reversed(files)`, rather than `files[::-1]` for speed

Reply:
Every little bit helps, i guess.  `reversed()` is 10x faster than `[::-1]`, though this code path is on a background thread not contributing to user response time...